productivity-suite/piercing-library/src/piercing-gateway.ts

```diff
@@ -330,19 +353,41 @@ export class PiercingGateway<Env> {
     const service = this.getFragmentFetcher(env, fragmentConfig.fragmentId);
     const newRequest = this.getRequestForFragment(request, fragmentConfig, env);
     const response = await service.fetch(newRequest);
-    const fragmentStream = response.body!;
-
-    let preFragment = `<piercing-fragment-host fragment-id=${fragmentConfig.fragmentId}>`;
-    const postFragment = "</piercing-fragment-host>";
-
-    if (prePiercing) {
-      preFragment = `
-                ${preFragment}
-                <style>
-                    ${fragmentConfig.prePiercingStyles}
-                </style>`;
+    let fragmentStream = response.body!;
+
+    const fragmentId = fragmentConfig.fragmentId;
+    const framework = fragmentConfig.framework;
+
+    const prePiercingStyles = prePiercing
+      ? `<style>${fragmentConfig.prePiercingStyles}</style>`
+      : ``;
+
+    let template = `
+      <piercing-fragment-host fragment-id=${fragmentConfig.fragmentId}>
+        ${prePiercingStyles}
+        --FRAGMENT_CONTENT--
```

Comment:
Agreed it's a little weird, but it's just a delimiter we use to split the fragment on, so we can make it anything really. It's place in the world is similar to something you'd see in a server side templating language, like `<% fragment_content %>`

I chose this because I kinda wanted it to stick out as specifically not HTML. I was worried that folks might confuse it for actual markup that would be on the page. `FRAGMENT_CONTENT` could be a constant, and #FRAGMENT_CONTENT might conceivably be a token found in `prePiercingStyles`, which would cause issues. By using something that is invalid HTML, we run less of a risk that the `.split` call on template fails.

Reply:
I see, yeah that makes sense, thanks for the clarification 🙂